centreon/src/Core/TimePeriod/Infrastructure/Repository/DbReadTimePeriodRepository.php

```diff
@@ -0,0 +1,316 @@
+                return $result['tp_id'] != $timePeriodId;
+            }
+            return false;
+        } else {
+            return ! (empty($result));
+        }
+    }
+
+    /**
+     * @param list<TimePeriod> $timePeriods
+     *
+     * @return void
+     *
+     * @throws AssertionFailedException
+     * @throws TimeRangeException
+     * @throws \PDOException
+     */
+    private function addExtraTimePeriods(array $timePeriods): void
+    {
+        if ($timePeriods === []) {
+            return;
+        }
+        $timePeriodIds = array_keys($timePeriods);
+        $timePeriodIncludeRequest = str_repeat('?, ', count($timePeriodIds) - 1) . '?';
+        $requestTemplates = $this->translateDbName(
+            <<<SQL
+            SELECT *
+            FROM `:db`.timeperiod_exceptions
+            WHERE timeperiod_id IN ($timePeriodIncludeRequest)
+            ORDER BY timeperiod_id ASC, exception_id ASC
+            SQL
+        );
+        $statement = $this->db->prepare($requestTemplates);
+        $statement->execute($timePeriodIds);
+
+        while (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{exception_id: int, timeperiod_id: int, days: string, timerange: string} $result
+             */
+            $timePeriods[$result["timeperiod_id"]]->addExtraTimePeriod(
+                new ExtraTimePeriod(
+                    $result['exception_id'],
+                    $result["days"],
+                    new TimeRange($result["timerange"])
+                )
+            );
+        }
+    }
+
+    /**
+     * @param list<TimePeriod> $timePeriods
+     *
+     * @return void
+     *
+     * @throws \PDOException
+     */
+    private function addTemplates(array $timePeriods): void
+    {
+        if ($timePeriods === []) {
+            return;
+        }
+        $timePeriodIds = array_keys($timePeriods);
+        $timePeriodIncludeRequest = str_repeat('?, ', count($timePeriodIds) - 1) . '?';
+        $requestTemplates = $this->translateDbName(
+            <<<SQL
+            SELECT rel.timeperiod_id, tp.tp_id, tp.tp_alias
+            FROM `:db`.timeperiod tp
+            INNER JOIN `:db`.timeperiod_include_relations rel
+              ON rel.timeperiod_include_id = tp.tp_id
+            WHERE rel.timeperiod_id IN ($timePeriodIncludeRequest)
+            ORDER BY rel.timeperiod_id ASC, rel.include_id ASC
+            SQL
+        );
+        $statement = $this->db->prepare($requestTemplates);
+        $statement->execute($timePeriodIds);
+
+        while (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{tp_id: int, tp_alias: string, timeperiod_id: int} $result
+             */
+            $timePeriods[$result["timeperiod_id"]]->addTemplate(
+                new Template($result["tp_id"], $result["tp_alias"])
+            );
+        }
+    }
+
+    /**
+     * @param array{
+     *     tp_id: int,
```

Comment:
Up to you, cf. idea of [@phpstan-type](https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases)